src/NServiceBus.Core/Serialization/SerializationFeature.cs

```diff
@@ -18,6 +18,15 @@ public SerializationFeature()
             Defaults(s =>
             {
                 s.SetDefault("AdditionalDeserializers", new List<SerializationDefinition>());
+                var conventions = s.Get<Conventions>();
+                var knownMessages = s.GetAvailableTypes().Where(conventions.IsMessageType);
+
+                var messageMetadataRegistry = new MessageMetadataRegistry(conventions);
+                foreach (var msg in knownMessages)
+                {
+                    messageMetadataRegistry.RegisterMessageType(msg);
+                }
+                s.Set<MessageMetadataRegistry>(messageMetadataRegistry);
```

Comment:
I'm on the fence about this block here. This seems to be code which would belong to the `Setup` logic of a feature since it doesn't really configure some trivial default values (it even uses `Set` instead of `SetDefault`). So I wonder whether we should move that back to Setup and instead have the `RoutingFeature` `DependsOn` this feature. I think that would make things more clear since they have this dependency anyway and DependsOn would make it clear instead of hiding it behind DI or feature initialization logic?